### PR TITLE
Refactor error raising to remove duplication

### PR DESCRIPTION
**Pull request by hugovk**

### CODE REVIEW

**Comment by hugovk** [comment]
This will conflict with https://github.com/jazzband/tablib/pull/437. Shall we merge this before or after that one?

**Comment by claudep** [comment]
Depends on whether you want to be nice to Daniel :-)

File: src/tablib/formats/__init__.py [hunk 1]
```diff
@@ -134,7 +116,13 @@ def formats(self):
     def get_format(self, key):
         if key not in self._formats:
             if key in uninstalled_format_messages:
-                raise UnsupportedFormat(uninstalled_format_messages[key])
+                uninstalled_format_messages[key].update(key=key)
```
Comment (bluetech) [inline src/tablib/formats/__init__.py]: Instead of modifying a global, which is not so nice, I'd suggest to add `key=key` to the `format(...)`.

Comment (hugovk) [reply src/tablib/formats/__init__.py]: Thanks, updated in 2b364fcc20c7d07ebdbe0b932e7d2fbbcb3eb860.

**Comment by hugovk** [comment]
Rebased!

**Comment by claudep** [comment]
Could you just squash the first two commits (letting only the pre-commit stuff separate)?

**Comment by hugovk** [comment]
OK, I can do that later. But won't "Squash and merge" nullify it in any case?

image

**Comment by claudep** [comment]
Sure we could do that, but as the pyupgrade commit is not related to the main issue, I would have let it as a separate commit.

**Comment by hugovk** [comment]
OK to squash and merge instead?

**Comment by claudep** [comment]
Done :-) Thanks for the patch.

**Pull request merged**